### ISSUE TITLE
Travis failure because of Spotbugs

### ISSUE THREAD

**Issue by leventov** [apache/incubator-druid#8135]
```
[ERROR] Unread field: org.apache.druid.indexing.worker.config.WorkerConfig.intermediaryPartitionCleanupPeriodSec; should this field be static? [org.apache.druid.indexing.worker.config.WorkerConfig] At WorkerConfig.java:[line 49] SS_SHOULD_BE_STATIC
[ERROR] Unread field: org.apache.druid.indexing.worker.config.WorkerConfig.intermediaryPartitionDiscoveryPeriodSec; should this field be static? [org.apache.druid.indexing.worker.config.WorkerConfig] At WorkerConfig.java:[line 46] SS_SHOULD_BE_STATIC
[ERROR] Unread field: org.apache.druid.indexing.worker.config.WorkerConfig.version; should this field be static? [org.apache.druid.indexing.worker.config.WorkerConfig] At WorkerConfig.java:[line 38] SS_SHOULD_BE_STATIC
[INFO] 
```

Probably caused by #8073.

FYI @Fokko

**Comment by Fokko** [comment]
Thanks for letting me know. I can fix this in a hour, I'm currently traveling. Feel free to revert the commit as well. I can open a new PR.

**Comment by Fokko** [comment]
Here we go https://github.com/apache/incubator-druid/pull/8138

**Comment by clintropolis** [comment]
Since #8138 seems to be failing with the same issue, I think we should consider reverting  #8073 and then maybe could repurpose #8138 to trying to enable this again so we can unblock CI.

**Comment by jon-wei** [comment]
@clintropolis Reverting #8073 sounds good

**Comment by jihoonson** [comment]
Raised https://github.com/apache/incubator-druid/pull/8143. Tested it locally.

**Comment by clintropolis** [comment]
In that case we can skip reverting #8138 I think

**Comment by jihoonson** [comment]
Sorry, there was some mistakes I think. #8143 doesn't work. +1 for reverting #8073.

**Comment by asdf2014** [comment]
Perhaps adding `@SuppressWarnings("SS_SHOULD_BE_STATIC")` annotations to `WorkerConfig` can solve this problem. After opening a new spotbugs check in the future, we should rebase the latest master branch and ensure that CI can pass.

**Comment by asdf2014** [comment]
@Fokko In #8138, the `@SuppressWarnings` annotation used will not work and should be replaced with `@SuppressFBWarnings`.

**Comment by asdf2014** [comment]
> Despite our best efforts, the FSF has never considered the Apache License to be compatible with GPL version 2, ...

Oops.. Using `@SuppressFBWarnings` from `com.google.code.findbugs:annotations` or `com.github.spotbugs:spotbugs-annotations` will result in a license conflict issue, even if the latter has LGPL 2.1 license. More details: https://www.apache.org/licenses/GPL-compatibility.html

**Comment by Fokko** [comment]
@asdf2014 Yes, we cannot bundle the annotations with Druid, since the license is incompatible with Apache. Running the maven plugin is fine afaik, but making it part of the software itself is not allowed.

**Comment by asdf2014** [comment]
> @asdf2014 Yes, we cannot bundle the annotations with Druid, since the license is incompatible with Apache. Running the maven plugin is fine afaik, but making it part of the software itself is not allowed.

![](https://i.loli.net/2019/07/24/5d382947aa7f716781.jpg)

**Comment by Fokko** [comment]
?

**Comment by asdf2014** [comment]
@Fokko Nothing, just that I feel very helpless. :sweat_smile:
